generator/ServiceClientGeneratorLib/Generators/SourceFiles/ServiceInterfaceNetStandard.tt

```diff
@@ -84,7 +84,7 @@ namespace <#=this.Config.Namespace#>
         Task<<#=operation.Name#>Response> <#=operation.Name#>Async(<#=operation.Name#>Request request, System.Threading.CancellationToken cancellationToken = default(CancellationToken));
 
 <#
-        if (this.Config.ServiceModel.H2Support == H2SupportDegree.EventStream && operation.IsEventStreamOutput)
+        if ((this.Config.ServiceModel.H2Support == H2SupportDegree.Optional && operation.IsEventStreamBidi) || (this.Config.ServiceModel.H2Support == H2SupportDegree.EventStream && operation.IsEventStreamOutput))
```

Comment:
Problem with my original logic as well but we should be checking if `H2Support` is set to `Required` and if so add the `#if NET8_0_OR_GREATER`.